app/rust/src/parser/message.rs

```diff
@@ -118,7 +118,11 @@ impl<'a> ByteString<'a> {
         });
 
         let mut copy_len = if self.0.len() > MAX_ASCII_LEN {
-            msg[MAX_ASCII_LEN..MAX_ASCII_LEN + suffix.len()].copy_from_slice(&suffix[..]);
+            if let Some(m) = msg.get_mut(MAX_ASCII_LEN..MAX_ASCII_LEN + suffix.len()) {
+                suffix.iter().zip(m.iter_mut()).for_each(|(s, m)| *m = *s);
```

Comment:
still iter_mut?

Reply:
well the gain could be not that much if replaced with copy_from_slice comparing with what we could get after using abort as panics. So I will replace them with copy_from_slice.